scripts/generate_index.py

```
@@ -0,0 +1,123 @@
+    if ext in EXT_TO_LANG:
+        return EXT_TO_LANG[ext]
+    # handle double suffix like .c++ or .c++ is not common; fallback to folder hints
+    parent = path.parent.name.lower()
+    if parent in ('c', 'c++', 'cpp'):
+        return 'C/C++'
+    return ext.lstrip('.') or 'unknown'
+
+
+def gather_files(root: Path) -> list[tuple[Path, str]]:
+    files = []
+    for dirpath, dirnames, filenames in os.walk(root):
```

Ruff (0.14.1)

Loop control variable `dirnames` not used within loop body

Rename unused `dirnames` to `_dirnames`

(B007)